tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -319,32 +291,32 @@ df_pca[!, :type] = repeat([1, 2]; inner=n_cells ÷ 2)
 ```
 
 We see the two groups are well separated in this 2-D space. As an unsupervised learning method, performing PCA on this dataset gives membership for each cell instance.
-Another way to put it, 2 dimensions is enough to display the main structure of the data.
+Another way to put it: 2 dimensions is enough to capture the main structure of the data.
 
 #### Further extension: automatic choice of the number of principal components with ARD
 
-A direct question arises from above practice is: how many principal components do we want to keep in order to sufficiently represent the latent structure in the data?
+A direct question arises from above practice is: how many principal components do we want to keep, in order to sufficiently represent the latent structure in the data?
 This is a very central question for all latent factor models, i.e. how many dimensions are needed to represent that data in the latent space.
-In the case of PCA, there exist a lot of heuristics to make that choice. For example, We can tune the number of principal components using empirical methods such as cross-validation based some criteria such as MSE between the posterior predicted $\hat{x}$ and $x$ or the percentage of variation explained [3].
+In the case of PCA, there exist a lot of heuristics to make that choice. For example, We can tune the number of principal components using empirical methods such as cross-validation based some criteria such as MSE between the posterior predicted (e.g. mean predictions) data matrix and the original data matrix or the percentage of variation explained [3].
 
-For p-PCA, this can be done in an elegant and principled way, suing a technique called *Automatic Relevance Determination*(ARD).
+For p-PCA, this can be done in an elegant and principled way, using a technique called *Automatic Relevance Determination* (ARD).
 ARD can help pick the correct number of principal directions by regularizing the solution space using a parameterized, data-dependent prior distribution that effectively prunes away redundant or superfluous features [4].
-Essentially, we are using a specific prior over the factor loadings $w$ that allows us to prune away dimensions in the latent space. The prior is determined by a precision hyperparameter $\alpha$. Here, smaller values of $\alpha$ correspond to more important components.
+Essentially, we are using a specific prior over the factor loadings $\mathbf{W}$ that allows us to prune away dimensions in the latent space. The prior is determined by a precision hyperparameter $\alpha$. Here, smaller values of $\alpha$ correspond to more important components.
 You can find more details about this in e.g. [5].
 
 ```julia
 @model function pPCA_ARD(x, ::Type{TV}=Array{Float64}) where {TV}
-    # retrieve the dimension of input matrix x.
+    # Dimensionality of the problem.
     N, D = size(x)
```

Comment:
Now it becomes even more confusing: Here the dimensions are again different.

Reply:
Thanks for spotting this. As in a previous response, after a thorough review consistency is as follows: in data generation, the data matrix `mat_exp` is of dimension `no. of features (genes) x no. of instances (cells)` which is D=9 x N=60; and in the model definitions `pPCA(X, k)` and `pPCA_ARD(X)`, the input argument `X` is of dimension `no. of instances x no. of features` which is N=60 x D=9.